### PR TITLE
AF-2680 Dart2

### PR DESCRIPTION
**Pull request by robbecker-wf**
# Summary
Fixes for Dart 2
- use dart2_constant
- adjust dependencies
- simplify travis CI and run under dart 1 and 2
- fix tests to specify @TestOn('browser') and adjust mocking to work under dart 2

# Testing
 - [ ] CI passes
 - [ ] Manual testing : Link into a real app and make sure routing still works

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on HipChat: InfoSec Forum._

File: lib/providers/browser_history.dart [hunk 1]
```diff
@@ -8,23 +8,29 @@ class BrowserHistory implements HistoryProvider {
...
+  @override
   String get urlStub => '';
 
+  @override
   String get pageTitle =>
       _pageTitle ?? (_window.document as HtmlDocument).title;
 
-  void set pageTitle(String title) {
+  @override
+  set pageTitle(String title) {
```
Comment (robbecker-wf) [inline lib/providers/browser_history.dart:26]: no return value for a setter

File: lib/providers/browser_history.dart [hunk 2]
```diff
@@ -8,23 +8,29 @@ class BrowserHistory implements HistoryProvider {
     _window = windowImpl ?? window;
   }
 
+  @override
   Stream get onChange => _window.onPopState;
 
+  @override
```
Comment (robbecker-wf) [inline lib/providers/browser_history.dart:14]: you'll see a lot of @override for methods that should have them.

File: lib/providers/hash_history.dart [hunk 3]
```diff
@@ -8,22 +8,28 @@ class HashHistory implements HistoryProvider {
     _window = windowImpl ?? window;
   }
 
-  Stream get onChange => _window.onHashChange;
+  @override
+  Stream<Event> get onChange => _window.onHashChange;
```
Comment (robbecker-wf) [inline lib/providers/hash_history.dart:12]: Adding a type here. This is only used in an internal method in Router.

File: lib/providers/hash_history.dart [hunk 4]
```diff
@@ -55,9 +62,11 @@ class HashHistory implements HistoryProvider {
...
   }
 
+  @override
   void back() {
     _window.history.back();
   }
 
-  String _normalizeHash(String hash) => hash.isEmpty ? '' : hash.substring(1);
+  String _normalizeHash(String hash) =>
+      hash != null && hash.isNotEmpty ? hash.substring(1) : '';
```
Comment (robbecker-wf) [inline lib/providers/hash_history.dart:71]: fixed for unit tests where hash was passed in null.

File: lib/providers/memory_history.dart [hunk 5]
```diff
@@ -19,20 +19,26 @@ class MemoryHistory implements HistoryProvider {
...
 
+  @override
   String get urlStub => _namespace;
 
+  @override
   String get pageTitle => _pageTitle;
 
-  void set pageTitle(String title) {
+  @override
+  set pageTitle(String title) {
```
Comment (robbecker-wf) [inline lib/providers/memory_history.dart:35]: no return type on a setter

**Review by maxwellpeterson-wf — APPROVED** [review]

**Comment by robbecker-wf** [comment]
Closing .. this repo has been absorbed into w_router.

**Pull request closed without merging**